chaincode/input.go

```diff
@@ -31,42 +31,46 @@ var (
 
 // inputObjective is the representation of input args to register a Objective
 type inputObjective struct {
-	Name                      string           `validate:"required,gte=1,lte=100" json:"name"`
-	DescriptionHash           string           `validate:"required,len=64,hexadecimal" json:"descriptionHash"`
-	DescriptionStorageAddress string           `validate:"required,url" json:"descriptionStorageAddress"`
-	MetricsName               string           `validate:"required,gte=1,lte=100" json:"metricsName"`
-	MetricsHash               string           `validate:"required,len=64,hexadecimal" json:"metricsHash"`
-	MetricsStorageAddress     string           `validate:"required,url" json:"metricsStorageAddress"`
-	TestDataset               inputDataset     `validate:"omitempty" json:"testDataset"`
-	Permissions               inputPermissions `validate:"required" json:"permissions"`
+	Name                      string            `validate:"required,gte=1,lte=100" json:"name"`
+	DescriptionHash           string            `validate:"required,len=64,hexadecimal" json:"descriptionHash"`
+	DescriptionStorageAddress string            `validate:"required,url" json:"descriptionStorageAddress"`
+	MetricsName               string            `validate:"required,gte=1,lte=100" json:"metricsName"`
+	MetricsHash               string            `validate:"required,len=64,hexadecimal" json:"metricsHash"`
+	MetricsStorageAddress     string            `validate:"required,url" json:"metricsStorageAddress"`
+	TestDataset               inputDataset      `validate:"omitempty" json:"testDataset"`
+	Permissions               inputPermissions  `validate:"required" json:"permissions"`
+	Metadata                  map[string]string `validate:"omitempty,lte=100,dive,keys,lte=50,endkeys,lte=100" json:"metadata"`
 }
 
 // inputDataset is the representation in input args to register a dataset
 type inputDataset struct {
-	DataManagerKey string   `validate:"omitempty,len=64,hexadecimal" json:"dataManagerKey"`
-	DataSampleKeys []string `validate:"omitempty,dive,len=64,hexadecimal" json:"dataSampleKeys"`
+	DataManagerKey string             `validate:"omitempty,len=64,hexadecimal" json:"dataManagerKey"`
+	DataSampleKeys []string           `validate:"omitempty,dive,len=64,hexadecimal" json:"dataSampleKeys"`
+	Metadata       map[string]string  `validate:"omitempty,lte=100,dive,keys,lte=50,endkeys,lte=100" json:"metadata"`
```

Comment:
As far as I know, inputDataset represents an internal object and not a dataset asset. In the chaincode, the dataset asset is identified as `dataManager`.
@thibaultrobert could you confirm?

Reply:
Yes you right it's only use as a part of an `inputObjective`. It represent the combination of one `dataManager` and several `dataSample`.
Adding a Metadata field seems a bit overkill.